core-rust/core-api-server/src/core_api/handlers/stream_transactions.rs

```diff
@@ -772,3 +841,228 @@ fn to_api_direct_substate_id(
+        hash: to_api_notarized_transaction_hash(&user_hashes.notarized_transaction_hash),
+        hash_bech32m: to_api_hash_bech32m(context, &user_hashes.notarized_transaction_hash)?,
+        payload_hex,
+        signed_transaction_intent: Box::new(to_api_signed_transaction_intent_v2(
+            context,
+            signed_transaction_intent,
+            &user_hashes.transaction_intent_hash,
+            &user_hashes.signed_transaction_intent_hash,
+        )?),
+        notary_signature: Some(to_api_signature(&notary_signature.0)),
+    })
+}
+
+pub fn to_api_signed_transaction_intent_v2(
+    context: &MappingContext,
+    signed: &SignedTransactionIntentV2,
+    transaction_intent_hash: &TransactionIntentHash,
+    signed_transaction_intent_hash: &SignedTransactionIntentHash,
+) -> Result<models::SignedTransactionIntentV2, MappingError> {
+    let SignedTransactionIntentV2 {
+        transaction_intent,
+        transaction_intent_signatures,
+        non_root_subintent_signatures,
+    } = signed;
+    Ok(models::SignedTransactionIntentV2 {
+        hash: to_api_signed_transaction_intent_hash(signed_transaction_intent_hash),
+        hash_bech32m: to_api_hash_bech32m(context, signed_transaction_intent_hash)?,
+        transaction_intent: Box::new(to_api_transaction_intent_v2(
+            context,
+            transaction_intent,
+            transaction_intent_hash,
+        )?),
+        transaction_intent_signatures: Box::new(to_api_intent_signatures_v2(
+            transaction_intent_signatures.signatures.as_slice(),
+        )),
+        non_root_subintent_signatures: non_root_subintent_signatures
+            .by_subintent
+            .iter()
+            .map(|sigs| to_api_intent_signatures_v2(&sigs.signatures))
+            .collect(),
+    })
+}
+
+pub fn to_api_intent_signatures_v2(
+    intent_signatures: &[IntentSignatureV1],
+) -> models::IntentSignatures {
+    models::IntentSignatures {
+        signatures: intent_signatures
+            .iter()
+            .map(|s| to_api_signature_with_public_key(&s.0))
+            .collect(),
+    }
+}
+
+pub fn to_api_transaction_intent_v2(
+    context: &MappingContext,
+    transaction_intent: &TransactionIntentV2,
+    transaction_intent_hash: &TransactionIntentHash,
+) -> Result<models::TransactionIntentV2, MappingError> {
+    let TransactionIntentV2 {
+        transaction_header,
+        root_intent_core,
+        non_root_subintents,
+    } = transaction_intent;
+    Ok(models::TransactionIntentV2 {
+        hash: to_api_transaction_intent_hash(transaction_intent_hash),
+        hash_bech32m: to_api_hash_bech32m(context, transaction_intent_hash)?,
+        transaction_header: Box::new(to_api_transaction_header_v2(transaction_header)?),
+        root_intent_core: Box::new(to_api_intent_core_v2(context, root_intent_core)?),
+        non_root_subintents: non_root_subintents
+            .0
+            .iter()
+            // TODO:CUTTLEFISH - fix the subintent hashes
```

Comment:
Just a thought: slighlty annoyed that we need to caclualte the intent hash on-the-fly. Maybe we can store that.

Reply:
Yes that's my plan! And hence why I didn't implement a hacky recalculation here.

Reply:
(fixed now)